static/src/javascripts-legacy/projects/common/modules/experiments/tests/tailor-survey.js

```diff
@@ -0,0 +1,166 @@
+define([
+    'bean',
+    'fastdom',
+    'Promise',
+    'lodash/functions/debounce',
+    'common/utils/config',
+    'common/utils/cookies',
+    'common/utils/storage',
+    'common/utils/mediator',
+    'common/utils/fastdom-promise',
+    'common/utils/private-browsing',
+    'text!common/views/experiments/tailor-survey.html',
+    'common/utils/fetch-json',
+    'lodash/collections/forEach'
+], function (
+    bean,
+    fastdom,
+    Promise,
+    debounce,
+    config,
+    cookies,
+    storage,
+    mediator,
+    fastdomPromise,
+    privateBrowsing,
+    quickSurvey,
+    fetchJson,
+    forEach
+) {
+    return function () {
+        this.id = 'TailorSurvey';
+        this.start = '2017-01-25';
+        this.expiry = '2017-02-31';
+        this.author = 'Manlio';
+        this.description = 'Testing Tailor surveys';
+        this.audience = 0.01;
+        this.audienceOffset = 0;
+        this.successMeasure = 'We can show a survey on Frontend to regular users only (as decided by Tailor)';
+        this.audienceCriteria = 'All users';
+        this.dataLinkNames = 'Tailor survey';
+        this.idealOutcome = '';
+
+        this.canRun = function () {
+            return !(config.page.isAdvertisementFeature) &&
+                config.page.contentType === 'Article'
+        };
+
+        function callTailor(bwid) {
+            if (!bwid) {
+                // console.log('No cookie found :(');
+            }
+            else {
+                var endpoint = 'https://tailor.guardianapis.com/suggestions?browserId=' + bwid;
+                var request = fetchJson(endpoint, {
+                    type: 'json',
+                    method: 'get'
+                });
+                return request.then(function (body) {
+                    return {
+                        tailorResponse: body
+                    }
+                });
+            }
+        }
+
+        function renderQuickSurvey() {
+            var bwid = cookies.get('bwid');
+
+            return callTailor(bwid).then(function (response) {
+                var isRegular = response.tailorResponse.userDataForClient.regular;
+                return isRegular;
+            }).then(function(isRegular) {
```

Comment:
I would just put this in a single `.then` block unless you're planning to extract each step as a separate function.